pkg/mongoproxy/plugins/schema/types.go

```diff
@@ -491,7 +491,7 @@ func (c *CollectionField) Validate(ctx context.Context, v interface{}, denyUnkno
 		}
 	case OBJECT_ARRAY:
 		switch vTyped := v.(type) {
-		case bson.A:
+		case primitive.A:
```

Comment:
do we need to change all bson.A?
from the error msg, it's only for object array?
probably we can try `case bson.A, primitive.A:`?

Reply:
yea, this happened on all array types, also []string. 
Tried with `case bson.A, primitive.A:`, will show: `Duplicate case primitive.A` 
The reason is bson.A's implementation is primitive.A, but while doing case assertion, will not be treated as same.